metadata/src/main/java/org/apache/kafka/controller/BrokerHeartbeatManager.java

```diff
@@ -91,8 +102,15 @@ static class BrokerHeartbeatState {
          */
         private BrokerHeartbeatState next;
 
-        BrokerHeartbeatState(int id) {
+        BrokerHeartbeatState(int id, List<Uuid> directories) {
```

Comment:
I think we should just require people to pass a list here. It's easy enough to pass `Collections.emptyList` if needed.